### PR TITLE
opentelemetry: prepare for 0.16.0 release

### PR DESCRIPTION
**Pull request by jtescher**
Breaking changes in this release:

- Upgrade to `v0.17.0` of `opentelemetry` (#1853)
  For list of breaking changes in OpenTelemetry, see the [v0.17.0 changelog](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry/CHANGELOG.md#v0170).
- Disable the default `tracing-log` feature (#1869)

### CODE REVIEW

**Comment by jtescher** [comment]
@hawkw did we want to move forward with #1869 before merging this one?

**Comment by davidbarsky** [comment]
> @hawkw did we want to move forward with #1869 before merging this one?

I'm not Eliza, but yes.

File: tracing-opentelemetry/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+# 0.16.0 (January 30, 2022)
+
+### Breaking Changes
+
+- Upgrade to `v0.17.0` of `opentelemetry` (#1853)
+  For list of breaking changes in OpenTelemetry, see the
+  [v0.17.0 changelog](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry/CHANGELOG.md#v0170).
+
```
Comment (davidbarsky) [inline tracing-opentelemetry/CHANGELOG.md:8]:
```suggestion

```

Comment (davidbarsky) [reply tracing-opentelemetry/CHANGELOG.md:8]:
```suggestion
- Disable the default `tracing-log` feature (#1869)
```

**Review by davidbarsky — APPROVED** [review]
If you accept my changes to the changelog, then we can merge and release this.

File: tracing-opentelemetry/CHANGELOG.md (hunk 1 above)
Comment (davidbarsky) [reply tracing-opentelemetry/CHANGELOG.md:8]: hrm—ignore the first suggested change. not sure why that's been added.

**Review by hawkw — CHANGES_REQUESTED** [review]
I just saw that #1869 just got merged, but I'm not sure if that was actually the correct solution to the problem. Let's take a minute to make sure that change was correct before we publish this release, please.

**Comment by jtescher** [comment]
<details>
  <summary>Previous Discord comment context</summary>

> Hi, is it possible to separate log and tracing?
> I tried disabling tracing-log but I cannot turn-off default-features of tracing-subscriber which is a dependncy of tracing-opentelemetry

```toml
tracing-opentelemetry = { version = "0.16.0", optional = true, default-features = false }
tracing-subscriber = { version = "0.3.6", default-features = false, features = [
    "smallvec",
    "std",
] }
```

```shell
$ cargo tree -i -p tracing-log
tracing-log v0.1.2
└── tracing-subscriber v0.3.6
    ├── mylib
    └── tracing-opentelemetry v0.16.0
        └── mylib
```
</details>

I cannot reproduce that example. E.g. This example `Cargo.toml`

```toml
[package]
name = "test"
version = "0.1.0"
edition = "2021"

[dependencies]
opentelemetry = "0.16.0"
tracing = "0.1.29"
tracing-opentelemetry = { version = "0.16.0", default-features = false }
tracing-subscriber = { version = "0.3.6", default-features = false, features = [ "smallvec", "std", ] }
```

`tracing-log` is not included if `default-features = false`.

```shell
$ cargo tree -i -p tracing-log
error: package ID specification `tracing-log` did not match any packages
```

#1869 was not to fix the above case, but to simply not require opting out explicitly.

**Review by hawkw — APPROVED** [review]

**Pull request merged**